### PR TITLE
CLN: Remove the engine parameter in CSVFormatter and to_csv

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

Internal code and #6581 indicate that this was long overdue.

### CODE REVIEW

File: doc/source/whatsnew/v0.19.0.txt [hunk 1]
```diff
@@ -40,6 +40,8 @@ Other enhancements
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+- ``DataFrame.to_csv()`` has dropped the ``engine`` parameter (:issue:`13419`)
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v0.19.0.txt]: Can you move this to "Removal of prior version deprecations/changes" (a bit below)

Comment (gfyoung) [inline doc/source/whatsnew/v0.19.0.txt]: Ah, good catch.  Fixed.

File: doc/source/whatsnew/v0.19.0.txt [hunk 2]
```diff
@@ -64,6 +64,7 @@ Deprecations
 Removal of prior version deprecations/changes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+- ``DataFrame.to_csv()`` has dropped the ``engine`` parameter (:issue:`13419`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.19.0.txt]: put the original issue number here (you can leave this one as well if you want).

**Comment by jreback** [comment]
see #11274 this was actually just deprecated in 0.17.0 (even though it *said* 0.13).

File: doc/source/whatsnew/v0.19.0.txt (hunk 2 above)
Comment (gfyoung) [inline doc/source/whatsnew/v0.19.0.txt]: Sure thing.  Done.

**Comment by jorisvandenbossche** [comment]
@gfyoung Master is now targeted for 0.19.0, so this can go in. Can you move the whatsnew note to the correct file?

**Comment by gfyoung** [comment]
@jreback , @jorisvandenbossche : Travis is passing, so ready to merge if there are no other concerns.

**Comment by jreback** [comment]
lgtm. ping on green.

**Comment by gfyoung** [comment]
@jreback : Travis is giving the green light.  Ready to merge if there are no other concerns.

**Comment by gfyoung** [comment]
@jreback : Travis is giving the green light.  Ready to merge if there are no other concerns.

**Comment by jreback** [comment]
@gfyoung yes, merging shortly.

**Comment by jreback** [comment]
thanks @gfyoung

**Pull request closed without merging**